src/tool/hpcrun/gpu/gpu-trace-channel.c

```diff
@@ -140,13 +143,27 @@ gpu_trace_channel_signal_consumer_when_full
 // interface functions
 //******************************************************************************
 
+struct thread_data_t *
+gpu_trace_channel_get_td(gpu_trace_channel_t *ch)
+{
+  return ch->td;
+}
+
+
+void
+gpu_trace_channel_set_td(gpu_trace_channel_t *ch, thread_data_t *td)
```

Comment:
this is a bad name since there is already a gpu_trace_channel_set. I think the td should get added to the trace channel when it is created rather than putting it in afterward. should it be passed to the constructor, or have the td acquired within the constructor?

Reply:
I am able to move gpu_trace_channel_set to:
gpu_trace_create -> gpu_trace_alloc -> gpu_trace_channel_alloc:

> channel->td = gpu_trace_stream_acquire(); <<< this is basically what gpu_trace_channel_set is doing so I can delete it.

However, I still need a getter function for getting the td out of channel. Is this name ok:
gpu_trace_channel_get_td () ?